File: php/features.php
<?php
declare(strict_types=1);

// HubspotCommunicationPreferences SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/DebugFeature.php';
require_once __DIR__ . '/feature/IdempotencyFeature.php';
require_once __DIR__ . '/feature/MetricsFeature.php';
require_once __DIR__ . '/feature/PagingFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class HubspotCommunicationPreferencesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new HubspotCommunicationPreferencesBaseFeature();
            case "debug":
                return new HubspotCommunicationPreferencesDebugFeature();
            case "idempotency":
                return new HubspotCommunicationPreferencesIdempotencyFeature();
            case "metrics":
                return new HubspotCommunicationPreferencesMetricsFeature();
            case "paging":
                return new HubspotCommunicationPreferencesPagingFeature();
            case "ratelimit":
                return new HubspotCommunicationPreferencesRatelimitFeature();
            case "retry":
                return new HubspotCommunicationPreferencesRetryFeature();
            case "test":
                return new HubspotCommunicationPreferencesTestFeature();
            case "timeout":
                return new HubspotCommunicationPreferencesTimeoutFeature();
            default:
                return new HubspotCommunicationPreferencesBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "debug":
            case "idempotency":
            case "metrics":
            case "paging":
            case "ratelimit":
            case "retry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}

File: py/hubspotcommunicationpreferences_sdk/features.py
# HubspotCommunicationPreferences SDK feature factory

from hubspotcommunicationpreferences_sdk.feature.base_feature import HubspotCommunicationPreferencesBaseFeature
from hubspotcommunicationpreferences_sdk.feature.debug_feature import HubspotCommunicationPreferencesDebugFeature
from hubspotcommunicationpreferences_sdk.feature.idempotency_feature import HubspotCommunicationPreferencesIdempotencyFeature
from hubspotcommunicationpreferences_sdk.feature.metrics_feature import HubspotCommunicationPreferencesMetricsFeature
from hubspotcommunicationpreferences_sdk.feature.paging_feature import HubspotCommunicationPreferencesPagingFeature
from hubspotcommunicationpreferences_sdk.feature.ratelimit_feature import HubspotCommunicationPreferencesRatelimitFeature
from hubspotcommunicationpreferences_sdk.feature.retry_feature import HubspotCommunicationPreferencesRetryFeature
from hubspotcommunicationpreferences_sdk.feature.test_feature import HubspotCommunicationPreferencesTestFeature
from hubspotcommunicationpreferences_sdk.feature.timeout_feature import HubspotCommunicationPreferencesTimeoutFeature


_FEATURES = {
    "base": lambda: HubspotCommunicationPreferencesBaseFeature(),
    "debug": lambda: HubspotCommunicationPreferencesDebugFeature(),
    "idempotency": lambda: HubspotCommunicationPreferencesIdempotencyFeature(),
    "metrics": lambda: HubspotCommunicationPreferencesMetricsFeature(),
    "paging": lambda: HubspotCommunicationPreferencesPagingFeature(),
    "ratelimit": lambda: HubspotCommunicationPreferencesRatelimitFeature(),
    "retry": lambda: HubspotCommunicationPreferencesRetryFeature(),
    "test": lambda: HubspotCommunicationPreferencesTestFeature(),
    "timeout": lambda: HubspotCommunicationPreferencesTimeoutFeature(),
}


def _make_feature(name):
    factory = _FEATURES.get(name)
    if factory is not None:
        return factory()
    return _FEATURES["base"]()


# True when this SDK was generated with the named feature class - the
# constructor's tolerance for extend-carried features reads this (an
# active name with no generated class must not become a BaseFeature
# stray when an extend instance carries it).
def _has_feature(name):
    return name in _FEATURES
